coriolis/conductor/rpc/server.py

```diff
@@ -316,13 +316,21 @@ def execute_replica_tasks(self, ctxt, replica_id, shutdown_instances):
         execution.action = replica
 
         for instance in execution.action.instances:
-            validate_replica_inputs_task = self._create_task(
-                instance, constants.TASK_TYPE_VALIDATE_REPLICA_INPUTS,
+            validate_replica_inputs_source_task = self._create_task(
+                instance,
+                constants.TASK_TYPE_VALIDATE_REPLICA_INPUTS_SOURCE,
                 execution)
 
+            validate_replica_inputs_destination_task = self._create_task(
+                instance,
+                constants.TASK_TYPE_VALIDATE_REPLICA_INPUTS_DESTINATION,
+                execution,
+                depends_on=[validate_replica_inputs_source_task.id])
```

Comment:
I think these two can run in parallel. Does the TASK_TYPE_VALIDATE_REPLICA_INPUTS_DESTINATION require task info of TASK_TYPE_VALIDATE_REPLICA_INPUTS_SOURCE?

If not, you can remove the dependency on validate_replica_inputs_source_task

Reply:
"Does the TASK_TYPE_VALIDATE_REPLICA_INPUTS_DESTINATION require task info of TASK_TYPE_VALIDATE_REPLICA_INPUTS_SOURCE?"
Yes, the VM's export info is required off the source to validate some destination-related params (most notably being the `storage_mappings` to determine if the selected types for each disk exists and is usable)

Reply:
but isn't this supposed to just validate, and not mutate the task info?

Reply:
@gabriel-samfira 

As @aznashwan said, the VM export info is needed for destination params validation.

The `validate_migration_import_input` done as part of the destination-related params validation requires various info about the VM (such as os_type, vm_name, nics, disks). These infos are passed to `validate_migration_import_input` method with the `export_info` dict parameter.

If you feel that it might be cleaner (me & Nasu discussed about this), I can add an initial task to get the VM info details and set the `export_info`, and execute both source & destination validation in parallel using info from the `get_vm_info` initial task.

For now, I only noticed that `source_provider.validate_migration_export_input` returns the `export_info` and that's why I set `export_info` in the source validation task (avoiding an extra task to get the VM info), and added the destination validation dependency.

Reply:
"If you feel that it might be cleaner (me & Nasu discussed about this), I can add an initial task to get the VM info details and set the export_info, and execute both source & destination validation in parallel using info from the get_vm_info initial task."
There already is a dedicated GET_INSTANCE_INFO task which saves the export_info in the task_info as well.
If needed, you could either:
  1) delete the GET_INSTANCE_INFO task completely
  2) schedule it as the first task and then run the validation tasks in parllel

At this point I'd prolly go with the latter

Reply:
Alright, let's go with `#2`.